pkg/app/migrations_test.go

```diff
@@ -0,0 +1,329 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package app
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_migrateSchema010To020(t *testing.T) {
+	var src = &Spec010{
+		APIVersion:  "0.1.0",
+		Kind:        "ksonnet.io/app",
+		Name:        "test-migration",
+		Version:     "0.0.1",
+		Description: "test migration description",
+		Authors: []string{
```

Comment:
can't approve of these test cases hard enough

Reply:
😊